src/app/core/services/api.service.ts

```diff
@@ -1,25 +1,27 @@
-import { Observable, map, catchError, of } from 'rxjs';
+import { Observable, map, catchError } from 'rxjs';
 import { QuizCategory } from '../../shared/models/quiz-category.model';
 import { inject, Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { environment } from '../../../environments/environment';
+import { ErrorHandlerService } from './error-handler.service';
+import { hideSpinner } from '../../store/categories.store';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ApiService {
   private readonly categoriesUrl = environment.categoriesUrl;
   private readonly http = inject(HttpClient);
+  private readonly errorHandlerService = inject(ErrorHandlerService);
 
   fetchCategories(): Observable<QuizCategory[]> {
     return this.http
     .get<{ trivia_categories: QuizCategory[] }>(this.categoriesUrl)
     .pipe(
       map(response => response.trivia_categories),
       catchError((error) => {
-        // eslint-disable-next-line no-console
-        console.error('Error fetching categories:', error);
-        return of([]);
+        hideSpinner();
```

Comment:
usage of hideSpinner() here is confusing. fetchCategories() does not set spinner to "show" state, and it's not clear why it should be responsible for hiding it. 
I believe hideSpinner() should be invoked on another level

Reply:
your this.errorHandlerService.handleError throws error, therefore I think you can catch it on the level where you are using fetchCategories() and invoke hideSpinner() there.